website/cue/reference/releases/0.31.0.cue

```diff
@@ -0,0 +1,550 @@
+			description: """
+				The `component_received_event_bytes_total` and `component_sent_event_bytes_total`
+				internal metrics have been updated to use a new measure, "estimated JSON size", that
+				is an estimate of the size of the event were it encoded as JSON rather than the
+				"in-memory size" of the event, which is an implementation detail. See [the upgrade
+				guide](/highlights/2023-07-05-0-31-0-upgrade-guide#event_json_size) for more
+				details.
+				"""
+			breaking: true
+			pr_numbers: [17516, 17542]
+		},
+		{
+			type: "enhancement"
+			scopes: ["shutdown"]
+			description: """
+				Vector's graceful shutdown time limit is now configurable (via
+				`--graceful-shutdown-limit-secs`) and able to be disabled (via
+				`--no-graceful-shutdown-limit`). See the [CLI
+				docs](docs/reference/cli/) for more.
+				"""
+			pr_numbers: [17479]
+		},
+		{
+			type: "enhancement"
+			scopes: ["sinks"]
+			description: """
+				Support for `zstd` compression was added to sinks support compression.
+				"""
+			contributors: ["akoshchiy"]
+			pr_numbers: [17371]
+		},
+		{
+			type: "enhancement"
+			scopes: ["prometheus_remote_write sink"]
+			description: """
+				The `prometheus_remote_write` sink now supports `zstd` and `gzip` compression in
+				addition to `snappy` (the default).
+				"""
+			contributors: ["zamazan4ik"]
+			pr_numbers: [17334]
+		},
+		{
+			type: "enhancement"
+			scopes: ["journald source"]
+			description: """
+				The `journald` source now supports a `journal_namespace` option to restrict the namespace of the units that the source consumes logs from.
+				"""
+			pr_numbers: [17648]
+		},
+		{
+			type: "fix"
+			scopes: ["buffers"]
+			description: """
+				A disk buffer deadlock that occurred on start-up after certain crash conditions was
+				fixed.
+				"""
+			pr_numbers: [17657]
+		},
+		{
+			type: "enhancement"
+			scopes: ["codecs"]
+			description: """
+				The `gelf`, `native_json`, `syslog`, and `json` decoders (configurable as
+				`decoding.codec` on sources) now have corresponding options for lossy UTF-8
+				decoding via `decoding.<codec name>.lossy = true|false`. This can be used to
+				accept invalid UTF-8 where invalid characters are replaced before decoded.
+				"""
+			pr_numbers: [17628, 17680]
+		},
+		{
+			type: "fix"
+			scopes: ["http_client source"]
+			description: """
+				The `http_client` no longer corrupts binary data by always trying to interpret as UTF-8 bytes. Instead options were added to encoders for lossy UTF-8 decoding (see above entry).
+				"""
+			pr_numbers: [17655]
+		},
+		{
+			type: "enhancement"
+			scopes: ["aws_kinesis_firehose sink", "aws_kinesis sink"]
+			description: """
+				The `aws_kinesis_firehose` and `aws_kinesis` sinks are now able to retry requests
+				with partial failures by setting `request_retry_partial` to true. The default is
+				`false` to avoid writing duplicate data if proper event idempotency is not in place.
+				"""
```

Comment:
```suggestion
			scopes: ["aws_kinesis_firehose sink", "aws_kinesis_streams sink"]
			description: """
				The `aws_kinesis_firehose` and `aws_kinesis_streams ` sinks are now able to retry requests
				with partial failures by setting `request_retry_partial` to true. The default is
				`false` to avoid writing duplicate data if proper event idempotency is not in place.
				"""
```